thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnalysisModel.js

```diff
@@ -1,24 +1,44 @@
 function AnalysisModel() {
   this.metric;
+  this.metricName;
+  this.dataset;
   this.metricId;
   this.timeRange;
+  this.dateRange;
   this.granularity;
   this.dimension;
   this.filters;
   this.currentStart;
-  this.currentEndl;
+  this.currentEnd;
   this.baselineStart;
   this.baselineEnd;
+  this.compareMode;
+
 }
 
 AnalysisModel.prototype = {
-  init: function (params) {
+  init: function () {
```

Comment:
Please use syntax `init() { ... }` to be consistent with other changes in this PR.